src/util/signature_utils.js

```diff
@@ -372,9 +372,34 @@ function authorize_request_account_by_token(token, requesting_account, is_secret
 }
 
 
+// In case the object access_keys in requesting_account has the is_active property
+// and is_active is false we will not continue to work on the request and immediately throw an error
+// access_key_id_to_find is a string (not SensitiveString)
+// inside requesting_account the access keys ids are SensitiveStrings
+function check_access_key_status(access_key_id_to_find, requesting_account) {
```

Comment:
Not sure why we need this function being called separately from `authorize_request_account_by_token`, it would be best if it would call it inside to check the access key it uses for the signature checks.

Reply:
I will change it to be part of `authorize_request_account_by_token`.